packit/api.py

```diff
@@ -841,6 +841,13 @@ def sync_release(
             PackitException, if the version of the latest upstream release cannot be told.
             PackitException, if the upstream repo or dist-git is dirty.
         """
+
+        if use_downstream_specfile:
+            logger.info(
+                "Using the downstream specfile instead of the upstream one.",
+            )
+            self.up.set_specfile(self.dg.specfile)
```

Comment:
I understand the need for this but am worried this can be hard to maintain/read/test/... 

A few ideas just from the top of my head (But you've spent the time on this, not me,  so maybe there is no cleaner solution...;):
* Would it be possible to be able to be able to pass a specfile object to the affected methods?
  * It might help reduce the dependency on the upstream objects.
  * It might cause a spaghetti code...;(
* Hide this specfile resolution behind a property/function/...
* Factor out the release syncing code to a separate class and provide the parameterization when initialising the object.

Reply:
The code you are showing me was already there I have not added it. I have just moved it from the middle of the method to the top.
I agree that this code is not so nice, probably as you said having a class instead of a method could help here but I don't know if I would start with this refactoring now for something that was already there.
I would like to have something working with less changes as possible and then we can clean and refactoring it.